### PR TITLE
Exposed the national weather alerts response in the One Call API

### PR DESCRIPTION
**Pull request by troya2**
Added the feature, updated the relevant fixture, and added supporting specs.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 * [#21](https://github.com/dblock/open-weather-ruby-client/pull/21), [#20](https://github.com/dblock/open-weather-ruby-client/pull/20), [#19](https://github.com/dblock/open-weather-ruby-client/pull/19), [#18](https://github.com/dblock/open-weather-ruby-client/pull/18): Added support for Stations API - [@wasabigeek](https://github.com/wasabigeek).
 * [#22](https://github.com/dblock/open-weather-ruby-client/pull/23): Removed API version from `Config#endpoint` - [@dblock](https://github.com/dblock).
+* Exposed the national weather alerts response in the One Call API [@troya2](https://github.com/troya2)
```
Comment (dblock) [inline CHANGELOG.md:5]: Fix the format and danger will stop complaining. Missing a `-`, a period and a PR link in front.

File: lib/open_weather/models/one_call/alert.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  module Models
+    module OneCall
+      class Alert < Model
+        property 'sender_name'
+        property 'event'
+        property 'start', transform_with: ->(v) { Time.at(v).utc } # UTC
+        property 'end', transform_with: ->(v) { Time.at(v).utc } # UTC
+        property 'description'
+
+        def initialize(args = nil, options = {})
```
Comment (dblock) [inline lib/open_weather/models/one_call/alert.rb:13]: You don't need to override the constructor unless you're doing something extra here. Just remove it.

File: spec/fixtures/open_weather/one_call/lat_lon.yml [hunk 3]
```diff
@@ -95,7 +95,10 @@ http_interactions:
...
         sky","icon":"01d"}],"clouds":0,"uvi":10.01},{"dt":1589997600,"sunrise":1589973148,"sunset":1590023597,"temp":{"day":298.21,"min":287.71,"max":299.71,"night":290.03,"eve":298.59,"morn":287.71},"feels_like":{"day":299.68,"night":289.89,"eve":301.62,"morn":287.74},"pressure":1014,"humidity":61,"dew_point":290.18,"wind_speed":1.3,"wind_deg":71,"weather":[{"id":800,"main":"Clear","description":"clear
-        sky","icon":"01d"}],"clouds":0,"uvi":10.16}]}'
+        sky","icon":"01d"}],"clouds":0,"uvi":10.16}],"alerts": [{"sender_name": "NWS Tulsa (Eastern Oklahoma)","event": "Heat Advisory","start": 1597341600,"end": 1597366800,"description": "...HEAT ADVISORY REMAINS IN EFFECT FROM 1 PM THIS AFTERNOON TO\n8 PM CDT THIS EVENING...\n* WHAT...Heat index values of 105 to 109 degrees expected.\n* WHERE...Creek, Okfuskee, Okmulgee, McIntosh, Pittsburg,\nLatimer, Pushmataha, and Choctaw Counties.\n* WHEN...From 1 PM to 8 PM CDT Thursday.\n* IMPACTS...The combination of hot temperatures and high\nhumidity will combine to create a dangerous situation in which\nheat illnesses are possible."}]}'
```
Comment (dblock) [inline spec/fixtures/open_weather/one_call/lat_lon.yml:98]: We usually don't hand-craft VCR-recorded data. You should record a new separate VCR K7 .yml that has alerts data and write a separate unit test for that.

File: spec/open_weather/one_call/one_call_spec.rb [hunk 4]
```diff
@@ -7,23 +7,40 @@
...
     expect(data.hourly.first).to be_a OpenWeather::Models::OneCall::HourlyWeather
     expect(data.hourly.first.temp).to eq 295.55
+
+    # Daily
     expect(data.daily).to be_a Array
     expect(data.daily.size).to eq 8
     expect(data.daily.first).to be_a OpenWeather::Models::OneCall::DailyWeather
     expect(data.daily.first.temp.night).to eq 293.67
+
+    # Alerts
```
Comment (dblock) [inline spec/open_weather/one_call/one_call_spec.rb:35]: Record a new VCR and split this test, or re-record a new lat_lon.yml and update numbers in this one.

**Review by dblock — COMMENTED** [review]
Thanks! Made a bunch of comments, this is close.

**Comment by dblock** [comment]
Any interest in finishing this @troya2?

**Comment by troya2** [comment]
@dblock Thanks for the prod - looking at it now.  I haven't used VCR, so I'll have to work on that. The other items are easy.  Committing them now.

File: spec/open_weather/one_call/one_call_spec.rb (hunk 4 above)
Comment (troya2) [reply spec/open_weather/one_call/one_call_spec.rb:35]: Done - re-recorded in a location with current weather alerts.

**Comment by troya2** [comment]
@dblock I believe I've resolved all of your comments.  Let me know if not.  Happy new year!

File: CHANGELOG.md [hunk 5]
```diff
@@ -12,6 +12,8 @@
 * [#27](https://github.com/dblock/open-weather-ruby-client/pull/27): Removed default values for Faraday’s SSL settings ca_file and ca_path - [@sunny](https://github.com/sunny).
 * [#21](https://github.com/dblock/open-weather-ruby-client/pull/21), [#20](https://github.com/dblock/open-weather-ruby-client/pull/20), [#19](https://github.com/dblock/open-weather-ruby-client/pull/19), [#18](https://github.com/dblock/open-weather-ruby-client/pull/18): Added support for Stations API - [@wasabigeek](https://github.com/wasabigeek).
 * [#22](https://github.com/dblock/open-weather-ruby-client/pull/23): Removed API version from `Config#endpoint` - [@dblock](https://github.com/dblock).
+* [#25](https://github.com/dblock/open-weather-ruby-client/pull/25): Exposed the national weather alerts response in the One Call API [@troya2](https://github.com/troya2).
```
Comment (dblock) [inline CHANGELOG.md:15]: This should go to the 0.4.1 next release above, please. Rebase.

File: spec/open_weather/one_call/one_call_spec.rb [hunk 6]
```diff
@@ -6,24 +6,41 @@
...
+    # Daily
     expect(data.daily).to be_a Array
     expect(data.daily.size).to eq 8
     expect(data.daily.first).to be_a OpenWeather::Models::OneCall::DailyWeather
-    expect(data.daily.first.temp.night).to eq 293.67
+    expect(data.daily.first.temp.night).to eq 286.56
+
+    # Alerts
+    expect(data.alerts.first).to be_a OpenWeather::Models::OneCall::Alert
+    expect(data.alerts.first.sender_name).to eq 'NWS Phoenix (Central Arizona and California Desert)'
```
Comment (dblock) [inline spec/open_weather/one_call/one_call_spec.rb:37]: Nitpicking, but we could do this, which may be slightly more elegant.

```ruby
data.alerts.first.tap do |alert|
   expect(alert).to be_a
   expect(alert.sender_name).to eq ...
end
```

**Review by dblock — CHANGES_REQUESTED** [review]
CHANGELOG is in the wrong place, the rest looks good!

**Comment by dblock** [comment]
Thanks for finishing this up! Want to help with co-maintenance of this gem? Make the next release? Drop me a note to dblock at dblock dot org with your rubygems username if yes.

**Pull request merged**